doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,195 @@
+.. role:: html(raw)
+   :format: html
+
+Architectural overview
+======================
+
+PennyLane is a framework for optimization and machine learning of quantum and
+hybrid quantum-classical computations. The library provides a unified
+architecture for near-term quantum computing devices, supporting various
+quantum information paradigms.
```

Comment:
Really nice intro!

Reply:
as a matter of fact took some help from the [abstract](https://arxiv.org/pdf/1811.04968.pdf) here :smile:

Reply:
😆